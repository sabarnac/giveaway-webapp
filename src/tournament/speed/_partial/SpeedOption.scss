.speed-control__option {
  background: #ffffff;
  width: 4.75rem;
  height: auto;
  padding: 1rem;
  line-height: 1rem;
  margin: 0.5rem 0.15rem;
  text-transform: lowercase;
}
