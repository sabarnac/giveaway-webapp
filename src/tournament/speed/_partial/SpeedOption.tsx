import React from "react";
import { observer, inject } from "mobx-react";
import "./SpeedOption.scss";
import classNames from "classnames";
import Config from "../../../store/config/Config";

/**
 * Properties of the animation speed controller view React component.
 */
interface SpeedOptionProps {
  /** @ignore The application config. */
  config?: Config;
  /** The speed option value. */
  speed: number;
}

/**
 * React component for the animation speed controller.
 */
export default inject("config")(
  observer(
    (props: SpeedOptionProps): JSX.Element => (
      <button
        key={props.speed}
        className={classNames("speed-control__option", {
          "button-primary": props.speed === props.config!.speed,
        })}
        onClick={() => props.config!.setSpeed(props.speed)}
      >
        {`${props.speed}X`}
      </button>
    ),
  ),
);
