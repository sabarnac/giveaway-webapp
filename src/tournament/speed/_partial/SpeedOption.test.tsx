import React from "react";
import SpeedOption from "./SpeedOption";
import { create, ReactTestRenderer } from "react-test-renderer";

describe("SpeedOption Component.", () => {
  it("Matches snapshot.", () => {
    const component: ReactTestRenderer = create(<SpeedOption speed={1} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
