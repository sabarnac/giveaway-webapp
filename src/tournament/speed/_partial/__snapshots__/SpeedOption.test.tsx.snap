// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeedOption Component. Matches snapshot. 1`] = `
<button
  className="foobar__option button-primary"
  onClick={[Function]}
>
  1x
</button>
`;
